recipes/fast-dds/2.3.X/conanfile.py

```diff
@@ -93,6 +94,12 @@ def _patch_sources(self):
         for patch in self.conan_data["patches"][self.version]:
             tools.patch(**patch)
 
+    def validate(self):
+        # FIXME: Foonathan dependency currently not proper working with cross compile 
+        # see https://github.com/conan-io/conan-center-index/pull/7632#discussion_r730445887
+        if hasattr(self, "settings_build") and tools.cross_building(self):
+            raise ConanInvalidConfiguration("Cross building is not yet supported. Contributions are welcome.")
+
```

Comment:
to remove, there is already a validate() method

Reply:
yes - already adapted